src/StripesContext.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+const Context = React.createContext();
+
+export const StripesProvider = Context.Provider;
+export const StripesConsumer = Context.Consumer;
+
+export function withStripes(WrappedComponent) {
+  return class extends React.Component {
+    render(props) {
+      return (
+        <StripesConsumer>
+          {stripes => <WrappedComponent {...props} stripes={stripes} /> }
```

Comment:
Should we account for the case where there is no stripes context, or if required, then perhaps even put a PropType on our wrapper component to validate for the stripes object?

Reply:
`stripes` is not a prop in this `WithStripes` component; `stripes` is a prop to the WrappedComponent, and will have a PropType there.

Reply:
Ah, yes. I think what I was really getting at was: what do we do in the event that there isn't a stripes context.... will it be an exception, or merely undefined. In the event that it is undefined, should we just render the component unwrapped.

Reply:
If there is no provider above the consumer, no exception occurs. There will be PropTypes warning from the WrappedComponent. I don't see an advantage for changing rendering when stripes is undefined.

Reply:
The advantage is that you just pass the component without wrapping it. It saves slightly in render time, but also the tree looks like:

Reply:
> What do we do in the event that there isn't a stripes context.... will it be an exception, or merely undefined. In the event that it is undefined, should we just render the component unwrapped.

That's got to be an exception. The whole purpose of `withStripes` is to get a Stripes object. If there isn't one to be had, there's no value in ploughing on regardless, and delaying the inevitable error until something less diagnosable happens.